ios/browser/api/wallet/hd_keyring_ios.mm

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import "brave/ios/browser/api/wallet/hd_keyring_ios.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+#include "base/strings/sys_string_conversions.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_service.h"
+#include "brave/components/brave_wallet/browser/hd_keyring.h"
+
+@interface HDKeyringIOS () {
+  brave_wallet::HDKeyring* _keyring;  // NOT OWNED
+}
+@end
+
+@implementation HDKeyringIOS
+
+- (instancetype)initWithKeyring:(brave_wallet::HDKeyring*)keyring {
+  if ((self = [super init])) {
+    _keyring = keyring;
+  }
+  return self;
+}
+
+- (HDKeyringType)type {
+  return HDKeyringTypeForBraveWalletHDKeyringType(_keyring->type());
+}
+
+- (bool)isEmpty {
+  return _keyring->empty();
+}
+
+- (void)clearData {
+  _keyring->ClearData();
+}
+
+- (void)addAccount {
+  [self addAccounts:1];
+}
+
+- (void)addAccounts:(NSUInteger)numberOfAcconts {
+  _keyring->AddAccounts(static_cast<size_t>(numberOfAcconts));
+}
+
+- (void)removeAccountWithAddress:(NSString*)address {
+  _keyring->RemoveAccount(base::SysNSStringToUTF8(address));
+}
+
+- (NSData*)signedMessageForAddress:(NSString*)address message:(NSData*)message {
+  std::vector<uint8_t> bridgedMessage;
+  uint8_t* msg = (uint8_t*)(message.bytes);
```

Comment:
`C++-style cast` instead of `C-style` cast.
NIT: `std::uint8_t`

Reply:
Tried C++ style casts and couldn't get it building, what cast would you use here when `message.bytes` is a `void*`? (Tried `reinterpret_cast` and `static_cast`)

Reply:
`static_cast<uint8_t*>(message.bytes)` should work if `message.bytes` is a `void*`. If it is `const void*` then you have to cast to `const uint8_t*`

Reply:
Ah thats what I was missing, the `const` definitions. Thanks, that works